main.tf

```diff
@@ -54,34 +54,40 @@ resource "random_string" "password" {
   number  = true
 }
 
-data "template_file" "postgrestemplate" {
-  template = "${file("${path.module}/templates/postgres-paas.json")}"
+resource "azurerm_postgresql_server" "postgres-paas" {
+  name                = "${var.product}-${var.env}"
+  location            = "${var.location}"
+  resource_group_name = "${azurerm_resource_group.data-resourcegroup.name}"
+
+  administrator_login          = "${var.postgresql_user}"
+  administrator_login_password = "${random_string.password.result}"
+
+  sku_name   = "${var.sku_name}"
+  sku_tier   = "${var.sku_tier}"
+  version    = "${var.postgresql_version}"
+  storage_mb = "${var.storage_mb}"
+
+  backup_retention_days        = "${var.backup_retention_days}"
+  geo_redundant_backup_enabled = "${var.georedundant_backup}"
+
+  ssl_enforcement_enabled = "${var.ssl_enforcement}"
+
+  tags = "${var.common_tags}"
 }
 
-resource "azurerm_template_deployment" "postgres-paas" {
-  template_body       = "${data.template_file.postgrestemplate.rendered}"
-  name                = "${var.product}-${var.env}"
+resource "azurerm_postgresql_database" "postgres-db" {
+  name                = "${replace(var.database_name, "-", "")}"
   resource_group_name = "${azurerm_resource_group.data-resourcegroup.name}"
-  deployment_mode     = "Incremental"
-
-  parameters = {
-    administratorLogin         = "${var.postgresql_user}"
-    administratorLoginPassword = "${random_string.password.result}"
-    location                   = "${var.location}"
-    env                        = "${var.env}"
-    serverName                 = "${var.product}-${var.env}"
-    dbName                     = "${replace(var.database_name, "-", "")}"
-    skuName                    = "${var.sku_name}"
-    skuCapacity                = "${var.sku_capacity}"
-    skuTier                    = "${var.sku_tier}"
-    version                    = "${var.postgresql_version}"
-    skuSizeMB                  = "${var.storage_mb}"
-    sslEnforcement             = "${var.ssl_enforcement}"
-    backupRetentionDays        = "${var.backup_retention_days}"
-    geoRedundantBackup         = "${var.georedundant_backup}"
-    charset                    = "${var.charset}"
-    collation                  = "${var.collation}"
-    dbRules                    = "${base64encode(jsonencode(local.db_rules))}"
-    commonTags                 = "${base64encode(jsonencode(var.common_tags))}"
-  }
+  server_name         = "${var.product}-${var.env}"
+  charset             = "${var.charset}"
+  collation           = "${var.collation}"
 }
+
+resource "azurerm_postgresql_virtual_network_rule" "postgres-vnet-rule" {
+  count                                = "${length(local.db_rules)}"
```

Comment:
can we avoid count? it will recreate the rules after it if only one changes in the middle

Reply:
We will need to upgrade the version of Terraform on the Jenkins agents as current version does not support for_each, but can have a discussion to see if that is an option.

Reply:
there's a .terraform-version file in each repo which declares the version required, I believe the minimum is now 0.12, after the key vault module upgrade that everyone was forced to do.

Reply:
OK cool, it was failing on the terraform validation checks earlier as the version does not appear to be overridden in this repo, but I can now see in each of the app repo's it is. I will revert back to for_each rather than count and for now I can stick in a .terraform-version file to my branch. thanks for the help

Reply:
See the changes required for CI in a module:
https://github.com/hmcts/cnp-module-key-vault/pull/25